src/main/kotlin/com/jetbrains/snakecharm/lang/psi/SmkFileReference.kt

```diff
@@ -1,34 +1,74 @@
 package com.jetbrains.snakecharm.lang.psi
 
+import com.intellij.openapi.module.ModuleUtilCore
+import com.intellij.openapi.roots.ModuleRootManager
 import com.intellij.openapi.util.TextRange
 import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.openapi.vfs.VfsUtil
 import com.intellij.openapi.vfs.VirtualFileManager
 import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
 import com.intellij.psi.PsiManager
 import com.intellij.psi.PsiReferenceBase
 import com.jetbrains.python.psi.PyStringLiteralExpression
 
-class SmkFileReference(
-        element: PsiElement, textRange: TextRange
-) : PsiReferenceBase<PsiElement>(element, textRange) {
-    private val key: String = element.text.substring(textRange.startOffset, textRange.endOffset)
+open class SmkFileReference(
+        element: SMKWorkflowParameterListStatement, textRange: TextRange
+) : PsiReferenceBase<SMKWorkflowParameterListStatement>(element, textRange) {
+    private val key = element.text.substring(textRange.startOffset, textRange.endOffset)
```

Comment:
Here is also not clear - you get some element, some range. If you save text into a variable, why not to pass already evaluated `key` to reference?  Or do copy text range content and just use method to get it when needed.

Also `key` here means `path`? `key` sounds not clear, like keyword of section or smth else